Add vessel speed over ground [all tests ci]

NikiDab opened the pull request:
This commit adds vessel speed over ground extracted from NMEA RMC, and VTG sentences.

leewujung commented:
@NikiDab : Just had a chance to look at the changes you made. In addition to the core function, make sure that you address all downstream or related changes in the package -- tests failing is a good indication that something that was working before doesn't run after the new changes. Also please add a test for the any new functionality or changes to ensure long-term reliability.

cc @jmjech

NikiDab commented:
@leewujung: Thanks for your comments. I have addressed all the comments except adding a test for the new functionality. I need a few clarifications on this: 

1) Should I place the .raw data in a Google Drive folder or use Git LFS? The file size that Gavin has provided is 36MB.

2) Should I add the test data name/address that Gavin shared with me to the list of other test examples in ECHOPYPE/echopype/test_data/README.md?

Thanks.

NikiDab commented:
@leewujung

1) I added code for parsing vessel speed from NMEA RMC and VTG sentences. There are no errors related to this part of the code.

2) I also added a test for this functionality. Currently, the test reads from a .raw file located in my local directory, which is causing the errors. These errors are expected and only occur because the test is referencing a custom path on my machine.

leewujung commented:
Closed and reopened to trigger all tests.

leewujung commented:
@NikiDab : Could you revert you changes on "@NikiDab
[Make apply_mask agnostic to the order of dimensions.](https://github.com/OSOceanAcoustics/echopype/pull/1482/commits/d372edf90546d18b50edc8661d95128e0dc81ede)" and anything related to that PR? Each PR should be standalone/singular when possible so that it is easier to figure things out when something fails. The changes in this and the other PR should be completely independent.

NikiDab commented:
@leewujung: Yes, that's correct. I thought I had removed it; I will fix it by the end of tomorrow.

NikiDab commented:
@leewujung: I think I was able to revert and remove the irrelevant data from this PR. Please let me know if this is not correct.

leewujung commented:
@NikiDab : Please fix the pre-commit errors and the failed tests. If there are downstream tests that fail due to your changes, those need to get fixed too. Thanks.

NikiDab commented:
@leewujung: Below, I have copied one of the error messages; it is raised due to the test function I have added named "test_ek80_speed_over_ground". As you can see, the error says "Expecting a .raw file but got". My understanding is that the .raw file is not placed in the right address. Could you please let me know; and if that is the case, could you please check to see if the files is placed in the proper location/address? Thanks.

[Running all Tests: echopype/tests/convert/test_convert_ek80.py#L657](https://github.com/OSOceanAcoustics/echopype/pull/1482/files#annotation_34111675701)
test_ek80_speed_over_ground ValueError: Expecting a .raw file but got

leewujung commented:
@NikiDab : The file is in the test_data/ek80/vessel_speed. You should be able to replicate the exact directory locally. Can you make sure your tests run without problem locally? 

Also, I see what you have `None` in the expected data, but in numerical arrays the empty values should be `NaN`. Please check the appropriate data type, and what's allowed in xarray datasets/dataarrays.

NikiDab commented:
@leewujung Thank you for your guides and feedback. I will follow those guidelines to resolve all the errors. Thanks.

NikiDab commented:
@leewujung: Hi Wu-Jung, I have been busy for the last two weeks. I will look into this during this week. Thanks.

leewujung commented:
@NikiDab : This has dragged on for a long time. If you are not able to finish this this week I'll ask someone else to help.

NikiDab commented:
@leewujung: Overall, there were six errors; I have been able to resolve two of them, which were related to the code I had added. The remaining four errors belong to a single category, I guess; meaning if one gets resolved, the rest of them can be solved easily. Below, I copied two of the errors. Any suggestions on how to approach it? 


<img width="1219" alt="Screenshot 2025-06-04 at 8 03 28 PM" src="https://github.com/user-attachments/assets/15a2bfb4-a139-4f4a-8eb3-8ef4351682a4" />

leewujung commented:
@NikiDab : I am afraid what you pasted does not give enough info. I suggest you step into the code to see what caused the errors.

leewujung commented:
@NikiDab: you should have access to the .raw files if you have set up the dev environment with docker as explained in the documentation. 

I’ll ask someone else to help with this.

leewujung commented:
Closing this as it has staled. Will be handled in a new PR.
